File: crates/dewasm-backend-java/units/rt/i64_rem_s.java
// requires: rt/trap
static long i64_rem_s(long a, long b) {
    if (b == 0) {
        trap("integer divide by zero");
    }
    if (a == Long.MIN_VALUE && b == -1) {
        return 0;
    }
    return a % b;
}

File: crates/dewasm-backend-ruby/units/rt/i64_rem_s.rb
# requires: rt/s64, rt/trap, rt/m64
def i64_rem_s(a, b)
  sa = s64(a)
  sb = s64(b)
  trap("integer divide by zero") if sb == 0
  r = sa.abs % sb.abs
  r = -r if sa < 0
  m64(r)
end
